### PR TITLE
feat: add support for docker build args to the task run command

### PR DESCRIPTION
**Pull request by briercliffe**
Adds the ability to pass build args into the docker build when running the `copilot task run` command.

Addresses #5376 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/cli/task_run.go [hunk 1]
```diff
@@ -1193,6 +1199,7 @@ func BuildTaskRunCmd() *cobra.Command {
 	cmd.Flags().StringVarP(&vars.groupName, taskGroupNameFlag, nameFlagShort, "", taskGroupFlagDescription)
 
 	cmd.Flags().StringVar(&vars.dockerfilePath, dockerFileFlag, defaultDockerfilePath, dockerFileFlagDescription)
+	cmd.Flags().StringToStringVar(&vars.dockerfileBuildArgs, dockerFileBuildArgsFlag, nil, dockerFileBuildArgsFlagDescription)
```
Comment (iamhopaul123) [inline internal/pkg/cli/task_run.go:1204]: Can you add an example to the [task run help menu] about how we can run task from dockerfile 🙏

**Review by iamhopaul123 — APPROVED** [review]
This is so much needed and the implementation is awesome! Thank you very much ❤️

File: internal/pkg/cli/flag.go [hunk 2]
```diff
@@ -192,6 +193,8 @@ var (
 	imageFlagDescription = fmt.Sprintf(`The location of an existing Docker image.
 Cannot be specified with --%s or --%s.`, dockerFileFlag, dockerFileContextFlag)
 	dockerFileFlagDescription = fmt.Sprintf(`Path to the Dockerfile.
+Cannot be specified with --%s.`, imageFlag)
+	dockerFileBuildArgsFlagDescription = fmt.Sprintf(`Key-value pairs converted to --build-args.
 Cannot be specified with --%s.`, imageFlag)
```
Comment (Lou1415926) [inline internal/pkg/cli/flag.go:198]: > Cannot be specified with --%s.`, imageFlag

aw yay thank you for adding this in the help menu message !

**Review by Lou1415926 — APPROVED** [review]
This looks great - thank you! I'm adding the DNM label for you to address iamhopaul123@'s comment!

File: internal/pkg/cli/task_run.go (hunk 1 above)
Comment (briercliffe) [reply internal/pkg/cli/task_run.go:1204]: I added an example to the help menu showing how to use the --build-args flag. Thanks for the suggestion!

File: internal/pkg/cli/task_run.go [hunk 3]
```diff
@@ -1181,7 +1181,9 @@ func BuildTaskRunCmd() *cobra.Command {
   Run a task using the current workspace with specific subnets and security groups.
   /code $ copilot task run --subnets subnet-123,subnet-456 --security-groups sg-123,sg-456
   Run a task with a command.
-  /code $ copilot task run --command "python migrate-script.py"`,
+  /code $ copilot task run --command "python migrate-script.py"
+  Run a task with Docker build-args.
```
Comment (Lou1415926) [inline internal/pkg/cli/task_run.go]:
```suggestion
  Run a task with Docker build args.
```

**Review by Lou1415926 — APPROVED** [review]
Thank you!!

**Pull request merged**